contracts/polygon/connectors/aave/v3-import/helpers.sol

```diff
@@ -231,22 +238,50 @@ contract AaveHelpers is Helper {
 		address[] memory tokens,
 		address userAccount
 	) internal {
-		for (uint256 i = 0; i < _length; i++) {
-			if (amts[i] > 0) {
-				uint256 _amt = amts[i];
-				require(
-					atokenContracts[i].transferFrom(
-						userAccount,
-						address(this),
-						_amt
-					),
-					"allowance?"
-				);
+		if (instaList.accountID(userAccount) == 0) {
+			for (uint256 i = 0; i < _length; i++) {
+				if (amts[i] > 0) {
+					uint256 _amt = amts[i];
+					require(
+						atokenContracts[i].transferFrom(
+							userAccount,
+							address(this),
+							_amt
+						),
+						"allowance?"
+					);
+					if (!getIsColl(tokens[i], address(this))) {
+						aave.setUserUseReserveAsCollateral(tokens[i], true);
+					}
+				}
+			}
+		} else {
+			string[] memory _targets = new string[](_length);
+			bytes[] memory _data = new bytes[](_length);
```

Comment:
@Richa-iitr I think first we need to check how many amts are greater than 0 and then assign the length.

Reply:
Done